### DISCUSSION TITLE
Can this library play videos?

### DISCUSSION THREAD

**Discussion by hestrro** [uwerat/qskinny#484] [Q&A] [answered]
Is it using qskgraphiclabel?

**Comment by uwerat** [comment, marked as answer]
In general this is the job of a multimedia framework - like f.e gstreamer.

Qt offers a wrapper around such frameworks known as Qt/Multimedia: https://doc.qt.io/qt-6/qtmultimedia-index.html
However the Qt/Quick APIs of Qt/Multimedia is only available from QML - even if everything that matters is implemented in C++.

QSkinny could offer a C++ API for the Qt/Quick Multimedia part - however using gstreamer without an extra wrapper seems to make more sense. Maybe have a look at https://gstreamer.freedesktop.org/documentation/qmlgl/qmlglsink.html ( hope this has nothing to do with QML despite its name )

See https://github.com/GStreamer/gst-plugins-good/blob/master/tests/examples/qt/qmlsink/main.cpp

**Reply by hestrro** [reply]
Maybe it would be  convenient to use Qt/Multimedia to implement it in a qml file, and then load it as a qquickitem into a QSkinny control via QQmlComponent component(&engine, QUrl(QStringLiteral("qrc:/qml.qml")));
